lib/puppet/provider/foreman_subnet/rest.rb

```diff
@@ -0,0 +1,263 @@
+Puppet::Type.type(:foreman_subnet).provide(:rest) do
+
+  confine :feature => :foreman_api
+
+  def subnets
+    ForemanApi::Resources::Subnet.new({
+      :base_url => resource[:base_url],
+      :oauth    => {
+        :consumer_key    => resource[:consumer_key],
+        :consumer_secret => resource[:consumer_secret]
+      }
+    }, {
+      :headers => {
+        :foreman_user => resource[:effective_user]
+      }
+    })
+  end
+
+  def subnet
```

Comment:
make it a private method, either `private :subnet` underneath or put it in a `private` block at the bottom of the class